libs/vkd3d-shader/spirv.c

```diff
@@ -5746,18 +5794,37 @@ static void vkd3d_dxbc_compiler_emit_resource_declaration(struct vkd3d_dxbc_comp
     }
     else
     {
-        type_id = vkd3d_dxbc_compiler_get_image_type_id(compiler,
-                reg, resource_type_info, sampled_type, structure_stride || raw, 0);
+        if (use_ssbo)
+        {
+            uint32_t array_type_id = vkd3d_spirv_build_op_type_runtime_array(builder,
+                    vkd3d_spirv_get_type_id(builder, VKD3D_TYPE_UINT, 1));
+            vkd3d_spirv_build_op_decorate1(builder, array_type_id, SpvDecorationArrayStride, 4);
+
+            type_id = vkd3d_spirv_build_op_type_struct(builder, &array_type_id, 1);
+            vkd3d_spirv_build_op_decorate(builder, type_id, SpvDecorationBufferBlock, NULL, 0);
+            vkd3d_spirv_build_op_member_decorate1(builder, type_id, 0, SpvDecorationOffset, 0);
+
+            if (!is_uav)
```

Comment:
Shouldn't this target the SSBO variable itself, not the member type?

Reply:
Offset has to be applied to a member anyway, and BufferBlock is applied to the struct type, not the variable. This matches glslang behaviour.

Reply:
Hm ... OK. Vulkan spec mentions that "If fragmentStoresAndAtomics is not enabled, then all storage image, storage texel buffer, and storage buffer variables in the fragment stage must be decorated with the NonWritable decoration." So in that case it needs to be set on the variable, but sure, this shouldn't really matter for us.